serialization/ruby/json/microsoft_kiota_serialization/lib/microsoft_kiota_serialization/json_parse_node.rb

```diff
@@ -0,0 +1,95 @@
+require 'microsoft_kiota_abstractions'
+require 'time'
+require 'json'
+require 'uuidtools'
+
+module MicrosoftKiotaSerialization
+  class JsonParseNode
+    include MicrosoftKiotaAbstractions::ParseNode
+    def initialize(node)
+      @current_node = node
+    end
+
+    def get_string_value
+      @current_node.to_s
+    end
+
+    def get_boolean_value
+      @current_node
+    end
+
+    def get_number_value
+      @current_node.to_i
+    end
+
+    def get_float_value
+      @current_node.to_f
+    end
+
+    def get_guid_value
+      UUIDTools::UUID.parse(@current_node)
+    end
+
+    def get_date_value
+      Time.parse(@current_node)
+    end
+
+    def get_collection_of_primitive_values(_type)
+      @current_node.map do |x|
+        current_parse_node = JsonParseNode.new(x)
+        begin
+          date = current_parse_node.get_date_value
+          date
+        rescue ArgumentError
+          begin
+            guid = current_parse_node.get_guid_value
+            guid
+          rescue ArgumentError
+            val = current_parse_node.get_string_value
+            val
+          end
+        end
+      end
+    end
+
+    def get_collection_of_object_values(type)
+      @current_node.map do |x|
+        current_parse_node = JsonParseNode.new(x)
+        val = current_parse_node.get_object_value(type)
+        val
+      end
+    end
+
+    def get_object_value(type)
+      item = type.new
+      assign_field_values(item)
+      item
+    rescue StandardError
+      raise StandardError, 'Error during deserialization'
```

Comment:
when you raise errors in ruby, do you have a way to pass the source error to the new one so the stack trace is maintained?
e.g. in java you'd do throw new RuntimeException("a message", sourceEx);